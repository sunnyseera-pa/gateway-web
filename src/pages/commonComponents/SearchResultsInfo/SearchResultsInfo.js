import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';

const SearchResultsInfo = ({ searchTerm, count }) => {
	const { t } = useTranslation();
	return (
		<>
			{searchTerm
				? t('searchResultsInfo.withTerm', { count: count, searchTerm: searchTerm })
				: t('searchResultsInfo.withOutTerm', { count: count })}
		</>
	);
};

SearchResultsInfo.propTypes = {
	searchTerm: PropTypes.string,
	count: PropTypes.number.isRequired,
};

export default SearchResultsInfo;
