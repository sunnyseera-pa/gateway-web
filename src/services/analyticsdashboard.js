import { useQuery } from 'react-query';
import { apiURL } from '../configs/url.config';
import { getRequest } from '../utils/requests';

const getTotalUsers = options => {
	return getRequest(`${apiURL}/analyticsdashboard/totalusers`, options);
};

const getUsersPerMonth = options => {
	return getRequest(`${apiURL}/analyticsdashboard/userspermonth`, options);
};

const useGetTotalUsers = (requestOptions, queryOptions = { queryKey: 'getTotalUsers' }) => {
	return useQuery({
		...queryOptions,
		queryFn: () => getTotalUsers(requestOptions),
	});
};

const useGetUsersPerMonth = (requestOptions, queryOptions = { queryKey: 'getUsersPerMonth' }) => {
	return useQuery({
		...queryOptions,
		queryFn: () => getUsersPerMonth(requestOptions),
	});
};

export default {
	getTotalUsers,
	getUsersPerMonth,
	useGetTotalUsers,
	useGetUsersPerMonth,
};
