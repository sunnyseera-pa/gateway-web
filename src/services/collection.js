import { useMutation, useQuery } from 'react-query';
import { apiURL } from '../configs/url.config';
import { getRequest, postRequest } from '../utils/requests';

export const getCollectionRequest = (_id, options) => {
	return getRequest(`${apiURL}/collections/${_id}`, options);
};

export const getCollectionRelatedObjectsRequest = (_id, options) => {
	return getRequest(`${apiURL}/collections/relatedobjects/${_id}`, options);
};

export const postCollectionCounterUpdateRequest = (data, options) => {
	return postRequest(`${apiURL}/collectioncounter/update`, data, options);
};

const useGetCollectionRequest = (_id, requestOptions, queryOptions = { queryKey: 'getCollectionRequest' }) => {
	return useQuery({
		...queryOptions,
		queryKey: [queryOptions.queryKey, _id],
		queryFn: async ({ queryKey }) => getCollectionRequest(queryKey[1], requestOptions),
	});
};

const useGetCollectionRelatedObjectsRequest = (_id, requestOptions, queryOptions = { queryKey: 'getCollectionRelatedObjectsRequest' }) => {
	return useQuery({
		...queryOptions,
		queryKey: [queryOptions.queryKey, _id],
		queryFn: async ({ queryKey }) => getCollectionRelatedObjectsRequest(queryKey[1], requestOptions),
	});
};

const usePostCollectionCounterUpdateRequest = (requestOptions, mutateOptions = { queryKey: 'postCollectionCounterUpdateRequest' }) => {
	return useMutation(data => postCollectionCounterUpdateRequest(data, requestOptions), {
		mutateOptions,
	});
};

export default {
	getCollectionRequest,
	getCollectionRelatedObjectsRequest,
	postCollectionCounterUpdateRequest,
	useGetCollectionRequest,
	useGetCollectionRelatedObjectsRequest,
	usePostCollectionCounterUpdateRequest,
};
