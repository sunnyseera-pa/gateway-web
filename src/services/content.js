import { useQuery } from 'react-query';
import { getRequest } from '../utils/requests';
import { UATCMSURL } from './content.constants';

const baseURL = require('../pages/commonComponents/BaseURL');
const cmsURL = baseURL.getCMSURL();
const env = baseURL.getURLEnv();
const local = 'local';

export const getNon5SafesModalContentRequest = (options) => {
    const url = env === local ? UATCMSURL : cmsURL;
	return getRequest(`${url}/Non5SafesModalContent`, options);
};

const useGetNon5SafesModalContentRequest = (requestOptions, queryOptions = { queryKey: 'getNon5SafesModalContentRequest' }) => {
	return useQuery({
		...queryOptions,
		queryFn: () => getNon5SafesModalContentRequest(requestOptions),
	});
};

export default {
    useGetNon5SafesModalContentRequest
}
