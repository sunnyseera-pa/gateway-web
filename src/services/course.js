import { useMutation, useQuery } from 'react-query';
import { apiURL } from '../configs/url.config';
import { getRequest, postRequest, putRequest } from '../utils/requests';

const getCourse = (_id, options) => {
	return getRequest(`${apiURL}/course/${_id}`, options);
};

const getEdit = (_id, options) => {
	return getRequest(`${apiURL}/course/edit/${_id}`, options);
};

const postCourse = (data, options) => {
	return postRequest(`${apiURL}/course`, data, options);
};

const putCourse = (_id, data, options) => {
	return putRequest(`${apiURL}/course/${_id}`, data, options);
};

const useGetCourse = (requestOptions, queryOptions = { queryKey: 'getCourse' }) => {
	return useQuery({
		...queryOptions,
		queryFn: _id => getCourse(_id, requestOptions),
	});
};

const useGetEdit = (requestOptions, queryOptions = { queryKey: 'getEdit' }) => {
	return useQuery({
		...queryOptions,
		queryFn: _id => getEdit(_id, requestOptions),
	});
};

const usePostCourse = (requestOptions, mutateOptions = { queryKey: 'postCourse' }) => {
	return useMutation(data => postCourse(data, requestOptions), {
		mutateOptions,
	});
};

const usePutCourse = (requestOptions, mutateOptions = { queryKey: 'putCourse' }) => {
	return useMutation((_id, data) => putCourse(_id, data, requestOptions), {
		mutateOptions,
	});
};

export default {
	getCourse,
	getEdit,
	postCourse,
	putCourse,
	useGetCourse,
	useGetEdit,
	usePostCourse,
	usePutCourse,
};
