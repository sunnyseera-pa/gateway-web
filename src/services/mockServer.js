import { setupServer } from 'msw/node';
import { rest } from 'msw';
import mswDatasets from './datasets/mockMsw';
import mswDatasetOnboarding from './dataset-onboarding/mockMsw';
import mswPostDatasetActivityLog from './activitylog/mockMsw';
import mswSearch from './search/mockMsw';
import mswGetLocations from './locations/mockMsw';
import mswGetContributors from './contributors/mockMsw';

import translations from '../../public/locales/en-GB/translation.json';

const mswGetEnTranslations = rest.get(`http://localhost/locales/en/translation.json`, (req, res, ctx) => {
	return res(ctx.status(200), ctx.json(translations));
});

const mswGetEnGbTranslations = rest.get(`http://localhost/locales/en-GB/translation.json`, (req, res, ctx) => {
	return res(ctx.status(200), ctx.json(translations));
});

const mswGetIcon = rest.get(`/images/Application_approved.svg`, (req, res, ctx) => {
	return res(ctx.status(200), ctx.text('<svg />'));
});

const handlers = [
	...mswDatasets,
	...mswDatasetOnboarding,
	...mswPostDatasetActivityLog,
	// ...mswSearch,
	...mswGetLocations,
	...mswGetContributors,
	mswGetEnTranslations,
	mswGetEnGbTranslations,
	mswGetIcon,
];

export const server = setupServer(...handlers);
