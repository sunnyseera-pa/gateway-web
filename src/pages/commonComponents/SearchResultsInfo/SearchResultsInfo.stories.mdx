import { Meta, Story, Canvas } from '@storybook/addon-docs';

import SearchResultsInfo from './SearchResultsInfo';

<Meta
	title='SearchResultsInfo'
	component={SearchResultsInfo}
	args={{
		count: 656,
		searchTerm: 'covid',
	}}
/>

<Canvas>
	<Story name='Basic'>{props => <SearchResultsInfo {...props} />}</Story>
</Canvas>
