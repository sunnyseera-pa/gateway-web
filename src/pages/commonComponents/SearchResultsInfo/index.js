import SearchResultsInfo from './SearchResultsInfo';

export default SearchResultsInfo;
