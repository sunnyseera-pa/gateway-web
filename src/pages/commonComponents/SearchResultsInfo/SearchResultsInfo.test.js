import { render, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import SearchResultsInfo from './SearchResultsInfo';
import { server } from '../../../services/mockServer';

let wrapper;
const props = {
	searchTerm: 'covid',
	count: 60,
};
describe('Given the SearchResultsInfo', () => {
	describe('When it is rendered', () => {
		beforeAll(() => {
			server.listen();
			wrapper = render(<SearchResultsInfo {...props} />, {
				wrapper: Providers,
			});
		});

		afterAll(() => {
			server.close();
		});

		it('Should match the snapshot', () => {
			expect(wrapper.container).toMatchSnapshot();
		});

		it('Then the results info should be displayed with search term', async () => {
			await waitFor(() => expect(wrapper.getByText("Showing 60 results for 'covid'")).toBeTruthy());
		});
	});
	describe('And when search term is empty', () => {
		beforeAll(() => {
			const { rerender } = wrapper;
			const newProps = {
				...props,
				searchTerm: '',
			};

			rerender(<SearchResultsInfo {...newProps} />, {
				wrapper: Providers,
			});
		});
		it('Then the results info should be displayed without search term', async () => {
			await waitFor(() => expect(wrapper.queryByText('Showing 60 results')).toBeTruthy());
		});
	});
});
