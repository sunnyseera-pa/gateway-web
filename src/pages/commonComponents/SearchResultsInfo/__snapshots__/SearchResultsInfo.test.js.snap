// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the SearchResultsInfo When it is rendered Should match the snapshot 1`] = `
<div>
  Showing 60 results for 'covid'
</div>
`;
